subprojects/core/src/main/java/org/gradle/internal/typeconversion/NotationParserBuilder.java

```diff
@@ -134,4 +134,17 @@ private NotationParserBuilder(TypeInfo<T> resultingType) {
 
         return new NotationConverterToNotationParserAdapter<Object, T>(composites.size() == 1 ? composites.get(0) : new CompositeNotationConverter<Object, T>(composites));
     }
+
+    private static class LazyDisplayName<T> {
```

Comment:
What would you use? Basically it works because I used an `Object` instead of a `String` for the display name in the delegating converter. If we create a `LazyDisplayName` class, then what else than `toString` would it implement? Which means it would itself take a `Factory<String>` to generate the lazy display name, and eventually 2 objects instead of one for the display name (`LazyDisplayName` + `Factory`).